File: Server/src/main/core/plugin/PluginManifest.java
package core.plugin;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface PluginManifest {
    public PluginType type() default PluginType.ACTION;
    public String name() default "";
}

File: Server/src/main/core/plugin/PluginManifest.kt
package core.plugin

/**
 * Plugin manifest.
 * @author Emperor
 *
 * @param type The type of the plugin (e.g. PluginType.ACTION)
 * @param name The name of the plugin.
 * @constructor Plugin manifest.
 */
@Retention(AnnotationRetention.RUNTIME)
annotation class PluginManifest(val type: PluginType = PluginType.ACTION, val name: String = "")
